src/Drawing.ts

```diff
@@ -0,0 +1,654 @@
+  _tag: 'Clipped',
+  shape,
+  drawing
+})
+
+/**
+ * Represents a `Drawing` that has had a shadow applied to it.
+ *
+ * @since 1.0.0
+ */
+export interface WithShadow {
+  readonly _tag: 'WithShadow'
+
+  /**
+   * The shadow to be applied.
+   */
+  readonly shadow: Shadow
+
+  /**
+   * The drawing to be shadowed.
+   */
+  readonly drawing: Drawing
+}
+
+/**
+ * Applies `Shadow` to a `Drawing`.
+ *
+ * @since 1.0.0
+ */
+export const withShadow: (shadow: Shadow, drawing: Drawing) => Drawing = (shadow, drawing) => ({
+  _tag: 'WithShadow',
+  shadow,
+  drawing
+})
+
+/**
+ * Gets a `Monoid` instance for `Drawing`.
+ *
+ * @since 1.0.0
+ */
+export const monoidDrawing: M.Monoid<Drawing> = {
+  concat: (x, y) =>
+    x._tag === 'Many' && y._tag === 'Many'
+      ? many(M.fold(readonlyArrayMonoidDrawing)([x.drawings, y.drawings]))
+      : x._tag === 'Many'
+      ? many(M.fold(readonlyArrayMonoidDrawing)([x.drawings, [y]]))
+      : y._tag === 'Many'
+      ? many(M.fold(readonlyArrayMonoidDrawing)([[x], y.drawings]))
+      : many([x, y]),
+  empty: many(readonlyArrayMonoidDrawing.empty)
+}
+
+const readOnlyArrayTraverseIO = ROA.readonlyArray.traverse(IO.io)
+
+const applyStyle: <A>(
+  oa: O.Option<A>,
+  f: (a: A) => (ctx: CanvasRenderingContext2D) => IO.IO<CanvasRenderingContext2D>
+) => (ctx: CanvasRenderingContext2D) => IO.IO<CanvasRenderingContext2D> = (fa, f) => (ctx) =>
+  pipe(
+    fa,
+    O.fold(
+      () => IO.of(ctx),
+      (a) => pipe(ctx, f(a))
+    )
+  )
+
+const renderShape: (shape: Shape) => (ctx: CanvasRenderingContext2D) => IO.IO<CanvasRenderingContext2D> = (shape) => {
+  switch (shape._tag) {
+    case 'Arc':
+      return C.arc(shape)
+
+    case 'Composite':
+      return (ctx) =>
+        pipe(
+          readOnlyArrayTraverseIO(shape.shapes, (s) => pipe(ctx, renderShape(s))),
+          IO.chain(() => IO.of(ctx))
+        )
+
+    case 'Path':
+      return (ctx) =>
+        pipe(
+          shape.points,
+          ROA.foldLeft(
+            () => IO.of(ctx),
+            (first, nexts) =>
+              pipe(
+                ctx,
+                C.moveTo(first),
+                IO.chain((ctx) => readOnlyArrayTraverseIO(nexts, (next) => pipe(ctx, C.lineTo(next)))),
+                IO.chain(() =>
+                  pipe(
+                    shape.closed,
+                    B.fold(
+                      () => IO.of(ctx),
+                      () => C.closePath(ctx)
+                    )
+                  )
+                )
+              )
+          )
+        )
+
+    case 'Rect':
+      return C.rect(shape)
+  }
+}
+
+/**
+ * Renders a `Drawing`.
+ *
+ * @since 1.0.0
+ */
+export const render: (drawing: Drawing) => (ctx: CanvasRenderingContext2D) => IO.IO<CanvasRenderingContext2D> = (
+  drawing
+) => {
+  const go: (drawing: Drawing) => (ctx: CanvasRenderingContext2D) => IO.IO<CanvasRenderingContext2D> = (drawing) => {
+    switch (drawing._tag) {
+      case 'Clipped':
+        return flow(
```

Comment:
There are some weird usages of `flow` (with only one argument) in this function

I mean

```
flow(f) = f
```

Reply:
Apologies, these must have been left over from when I was trying to clean up the nested arrow functions. I think I removed all of the extraneous calls to `flow`. See https://github.com/gcanti/graphics-ts/pull/7/commits/aaf3a44f1d61e72ab05ea1e2a41064c86089f2e1